kernels/volk/volk_32f_s32f_add_32f.h

```diff
@@ -0,0 +1,275 @@
+                                               const float* aVector,
+                                               const float scalar,
+                                               unsigned int num_points)
+{
+    unsigned int number = 0;
+    const unsigned int quarterPoints = num_points / 4;
+
+    float* cPtr = cVector;
+    const float* aPtr = aVector;
+
+    __m128 aVal, bVal, cVal;
+    bVal = _mm_set_ps1(scalar);
+    for (; number < quarterPoints; number++) {
+        aVal = _mm_loadu_ps(aPtr);
+
+        cVal = _mm_add_ps(aVal, bVal);
+
+        _mm_storeu_ps(cPtr, cVal); // Store the results back into the C container
+
+        aPtr += 4;
+        cPtr += 4;
+    }
+
+    number = quarterPoints * 4;
+    volk_32f_s32f_add_32f_generic(cPtr, aPtr, scalar, num_points - number);
+}
+#endif /* LV_HAVE_SSE */
+
+#ifdef LV_HAVE_AVX
+#include <immintrin.h>
+
+static inline void volk_32f_s32f_add_32f_u_avx(float* cVector,
+                                               const float* aVector,
+                                               const float scalar,
+                                               unsigned int num_points)
+{
+    unsigned int number = 0;
+    const unsigned int eighthPoints = num_points / 8;
+
+    float* cPtr = cVector;
+    const float* aPtr = aVector;
+
+    __m256 aVal, bVal, cVal;
+    bVal = _mm256_set1_ps(scalar);
+    for (; number < eighthPoints; number++) {
+
+        aVal = _mm256_loadu_ps(aPtr);
+
+        cVal = _mm256_add_ps(aVal, bVal);
+
+        _mm256_storeu_ps(cPtr, cVal); // Store the results back into the C container
+
+        aPtr += 8;
+        cPtr += 8;
+    }
+
+    number = eighthPoints * 8;
+    volk_32f_s32f_add_32f_generic(cPtr, aPtr, scalar, num_points - number);
+}
+#endif /* LV_HAVE_AVX */
+
+#ifdef LV_HAVE_NEON
+#include <arm_neon.h>
+
+static inline void volk_32f_s32f_add_32f_u_neon(float* cVector,
+                                                const float* aVector,
+                                                const float scalar,
+                                                unsigned int num_points)
+{
+    unsigned int number = 0;
+    const float* inputPtr = aVector;
+    float* outputPtr = cVector;
+    const unsigned int quarterPoints = num_points / 4;
+
+    float32x4_t aVal, cVal, scalarvec;
+
+    scalarvec = vdupq_n_f32(scalar);
+
+    for (number = 0; number < quarterPoints; number++) {
+        aVal = vld1q_f32(inputPtr);        // Load into NEON regs
+        cVal = vaddq_f32(aVal, scalarvec); // Do the add
+        vst1q_f32(outputPtr, cVal);        // Store results back to output
+        inputPtr += 4;
+        outputPtr += 4;
+    }
+
+    number = quarterPoints * 4;
+    volk_32f_s32f_add_32f_generic(cPtr, aPtr, scalar, num_points - number);
+}
+#endif /* LV_HAVE_NEON */
+
+
+#endif /* INCLUDED_volk_32f_s32f_add_32f_u_H */
+
+
+#ifndef INCLUDED_volk_32f_s32f_add_32f_a_H
+#define INCLUDED_volk_32f_s32f_add_32f_a_H
```

Comment:
The include guards should be consolidated into one. For me, this is mainly a matter of consistency and it would be great to have it in here.